src/Umbraco.Web/PublishedCache/XmlPublishedCache/XmlPublishedContent.cs

```diff
@@ -432,7 +433,20 @@ private void InitializeNode()
 		    }
 
             //lookup the content type and create the properties collection
-            contentType = getPublishedContentType(PublishedItemType.Content, docTypeAlias);
+            try
+            {
+                contentType = getPublishedContentType(PublishedItemType.Content, docTypeAlias);
+
+            }
+            catch (Exception e)
+            {
+                if (e.TargetSite.Name == "CreatePublishedContentType")
+                {
+                    content.Instance.RefreshContentFromDatabase();

```

Comment:
This will re-load the xml from the database in memory but it won't re-save the xml file so the next time the site restarts the same problem will remain. You also need to call `content.Instance.Commit(true)` here.